File: scripts/value_backup_scripts/counties/HarrisCounty.py
from utils.SpecialScraper import SpecialScript
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as EC

class HarrisCountyScript(SpecialScript):
    def perform_scraping(self):
        self.url = 'https://public.hcad.org/records/Real.asp'
        self.driver.get(self.url)

        acct_input = self.wait.until(EC.presence_of_element_located((By.ID, 'acct')))
        acct_input.send_keys(self.lookup_value)

        tax_year_select = self.wait.until(EC.presence_of_element_located((By.NAME, 'TaxYear')))
        for option in tax_year_select.find_elements(By.TAG_NAME, 'option'):
            if option.get_attribute('value') == self.year:
                option.click()
                break

        submit_button = self.wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, 'input[type="submit"][value="Search"]')))
        submit_button.click()

        print_link_element = self.wait.until(EC.presence_of_element_located((By.CSS_SELECTOR, 'td.subheader a[target="_blank"]')))

        self.print_link = print_link_element.get_attribute('href')

File: scripts/value_backup_scripts/counties/HarrisCounty.js
import BaseScript from '../../../core/BaseScript.js';

class HarrisCountyScript extends BaseScript {
  async performScraping() {
    await this.page.goto(this.accountLookupString, { waitUntil: 'networkidle2' });
    console.log(`Navigated to: ${this.page.url()}`);
  
    try {
      const acctInput = await this.page.waitForSelector('#acct', { visible: true, timeout: 5000 });
      await acctInput.type(this.account);
      
      const taxYearSelect = await this.page.waitForSelector('select[name="TaxYear"]', { visible: true, timeout: 5000 });
      const options = await taxYearSelect.$$('option');
      
      let yearSelected = false;
      for (const option of options) {
        const value = await option.evaluate(el => el.value);
        if (value === this.year) {
          await option.evaluate(b => b.click());
          yearSelected = true;
          break;
        }
      }
      
      const submitButton = await this.page.waitForSelector('input[type="submit"][value="Search"]', { visible: true, timeout: 10000 });
      await this.page.evaluate(button => button.click(), submitButton);
      
      const printLinkElement = await this.page.waitForSelector('td.subheader a[target="_blank"]', { visible: true, timeout: 10000 });
      this.printLink = await printLinkElement.evaluate(el => el.href);
    } catch (error) {
      console.error(`An error occurred: ${error.message}`);
    }
  }
}

export default HarrisCountyScript;
